openmls/src/binary_tree/array_representation/diff.rs

```diff
@@ -253,11 +220,12 @@ impl<'a, T: Clone + Debug> AbDiff<'a, T> {
         // is never empty, since if there is a second leaf index (that is within
         // the bound of the tree), there is a non-leaf root node that is in the
         // direct path of all leaves.
-        self.leaf_pair_check(leaf_index_1, leaf_index_2)?;
+        debug_assert!(leaf_index_1 != leaf_index_2);
+        debug_assert!(leaf_index_1.to_tree_index() < self.size);
+        debug_assert!(leaf_index_2.to_tree_index() < self.size);
```

Comment:
This is another place where we have a crate-public function where this PR replaces a runtime check with a debug-time one without getting any guarantee from the input type. I'd much prefer keeping the check here, potentially return an error and then hide the error at the same place where we do the actual check that these two indices don't collide.